### PR TITLE
Sync new format

### LINKED ISSUE #198: sync: produce `tests.toml` files that comply with the new spec
The updated spec for `tests.toml` files stores each test case as follows:

```toml
[<uuid>]
description = "<description>"
include = "<included>"
```

Each test case will be in its own table. With this format, we can selectively update the description and included value, while keeping any other properties left untouched.

The discussion that led to this format is in https://github.com/exercism/configlet/issues/186.

## Defaults

As the default case will be to include a test case, we only require `include` to be specified when its value is `false`. Therefore:

```toml
[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
description = "basic"
include = true
```

is equivalent to:

```toml
[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
description = "basic"
````

When writing new, included test cases, configlet will output them in the second format (without `include = true`). If configlet is updating an existing test case, it will retain the format used in the source file.

## Comments

Comments can be added as properties (not as regular comments):

```toml
[<uuid>]
description = "<description>"
include = "<included>"
comment = "<comment>"
```

If a track was to use multiple comment lines, they can do so using an array:

```toml
[<uuid>]
description = "<description>"
include = "<included>"
comments = [
    "<comment-line-1>",
    "<comment-line-2>",
    "<comment-line-3>"
]
```

## Heading

To prevent maintainers from adding comments that will then be removed when running `configlet sync` again, we'll add the following heading to the file:

```toml
# This is an auto-generated file. Regular comments will be removed when this
# file is regenerated. Regenerating will not touch any manually added keys,
# so comments can be added in a "comment" key.
```

## Example

```toml
# This is an auto-generated file. Regular comments will be removed when this
# file is regenerated. Regenerating will not touch any manually added keys,
# so comments can be added in a "comment" key.

[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
description = "basic"

[79ae3889-a5c0-4b01-baf0-232d31180c08]
description = "lowercase words"
include = false

[ec7000a7-3931-4a17-890e-33ca2073a548]
description = "invalid input"
include = false
comment = "excluded because we don't want to add error handling to the exercise"
```

## Conversion

We've sent [bulk PRs](https://github.com/exercism/v3-launch/issues/22) to all tracks to convert the existing `tests.toml` files.

We will sent a future update to remove the `include = true` statements.

### PR DESCRIPTION
**Pull request by joshiraez**
Fixes #198 
Closes https://github.com/exercism/configlet/issues/207

### CODE REVIEW

**Comment by joshiraez** [comment]
Is finished and I have been testing with enabled, disabled, the comments and stuff and it seems it is fine.

Thanks @ee7 for lending me a hand before :)

**Comment by ee7** [comment]
I believe that in the current state (f59451af049b7bc798c0b4d5572111fdcfb61b0b), `configlet sync` thinks that every existing track test case is missing.

There's also a end-to-end test that needs to be updated - I'll do it.

**Comment by ErikSchierboom** [comment]
I've commented on the original issue to help clarify things a bit: https://github.com/exercism/configlet/issues/198#issuecomment-828966838

**Comment by ee7** [comment]
With 1ef1e7eabca428c66ae0b4a39f7af8c4d0f71e0f, this input TOML:

```toml
[dd40c4d2-3c8b-44e5-992a-f42b393ec373]
description = "no matches"
include = true
str1 = """
Roses are red
Violets are blue"""

str2 = """\
  The quick brown \
  fox jumps over \
  the lazy dog.\
  """
lines = '''
The first newline is
trimmed in raw strings.
All other whitespace
is preserved.
'''
2 = "hi"
comments = ["first",
            "second",
             "third"]
```

is transformed to
```toml
[dd40c4d2-3c8b-44e5-992a-f42b393ec373]
description = "no matches"
str1 = """
Roses are red
Violets are blue"""
str2 = "The quick brown fox jumps over the lazy dog."
lines = """
The first newline is
trimmed in raw strings.
All other whitespace
is preserved.
"""
2 = "hi"
comments = ["first", "second", "third"]
```

Note that:
- `str2` becomes a single line
- `'''` becomes `"""`
- the `comments` array becomes a single line

We can't parse the TOML and from there output it again while preserving the style, because the parsed structure doesn't contain the style information.

Maybe we should output arrays as multi-line, even if they weren't multi-line previously.

But otherwise I think this is good enough for now.

@ErikSchierboom could you review and maybe play with it a little? Our test coverage isn't great here.

File: src/sync/tracks.nim [hunk 1]
```diff
@@ -61,14 +61,22 @@ proc newTrackExerciseTests(exercise: TrackRepoExercise): TrackExerciseTests =
...
+        let isIncluded = val["include"].getBool()
+        if isIncluded:
+          result.included.incl(uuid)
+        else:
+          result.excluded.incl(uuid)
+      else:
+        let msg = "Error: the value of an `include` key is `" &
+                  val["include"].toTomlString() & "`, but it must be a bool:\n" &
+                  exercise.testsFile()
+        stderr.writeLine(msg)
```
Comment (ee7) [inline src/sync/tracks.nim:77]: This should clearly use a helper proc that prints error messages, but we should handle that everywhere later.

**Review by ErikSchierboom — APPROVED** [review]
Lovely!

**Comment by ee7** [comment]
I'll just fix this PR for these, while I'm here:
https://github.com/exercism/erlang/blob/4cf8389c33fdc056ba1a019179e2a14fe3bbfc68/exercises/practice/grade-school/.meta/tests.toml#L12-L15
and
https://github.com/exercism/erlang/blob/4cf8389c33fdc056ba1a019179e2a14fe3bbfc68/exercises/practice/hamming/.meta/tests.toml#L36-L41

**Comment by ee7** [comment]
Okay,
```toml
[dd40c4d2-3c8b-44e5-992a-f42b393ec373]
description = "no matches"
include = true
str1 = """
Roses are red
Violets are blue"""

str2 = """\
  The quick brown \
  fox jumps over \
  the lazy dog.\
  """
lines = '''
The first newline is
trimmed in raw strings.
All other whitespace
is preserved.
'''
2 = "hi"
comments = ["first",
            "second",
             "third"]
```

now becomes

```toml
[dd40c4d2-3c8b-44e5-992a-f42b393ec373]
description = "no matches"
str1 = """
Roses are red
Violets are blue"""
str2 = "The quick brown fox jumps over the lazy dog."
lines = """
The first newline is
trimmed in raw strings.
All other whitespace
is preserved.
"""
2 = "hi"
comments = [
  "first",
  "second",
  "third",
]
```

**Pull request merged**